assessment-api/assessment-actors/src/main/scala/org/sunbird/actors/QuestionSetActor.scala

```diff
@@ -161,9 +163,50 @@ class QuestionSetActor @Inject()(implicit oec: OntologyEngineContext) extends Ba
 			DataNode.systemUpdate(request, response,"questionSet", Some(HierarchyManager.getHierarchy))
 		}).map(node => ResponseHandler.OK.put("identifier", identifier).put("status", "success"))
 	}
-
 	def copy(request: Request): Future[Response] ={
 		RequestUtil.restrictProperties(request)
 		CopyManager.copy(request)
 	}
+	def updateComment(request: Request): Future[Response] = {
+		AssessmentManager.getValidatedNodeForUpdateComment(request, "ERR_QUESTION_SET_UPDATE_COMMENT").flatMap(_ => updateInnerComment(request))
+	}
+	def updateInnerComment(request: Request): Future[Response] = {
+		val readReq = new Request(request)
+		val writeReq = new Request(request)
+
+		readReq.getRequest.remove("comments")
+		writeReq.getRequest.remove("comments")
+
+		val comments = request.getRequest.get("comments").asInstanceOf[java.util.ArrayList[java.util.Map[String, Object]]].asScala.toList
+		val futureResults = Future.traverse(comments) { comment =>
+			val identifier = comment.get("identifier").asInstanceOf[String]
+			val rejectComment = comment.get("comment")
+
+			readReq.getRequest.clear()
+			readReq.getRequest.put("rejectComment", rejectComment)
+			readReq.getContext.replace("identifier", identifier)
+
+			RequestUtil.validateRequest(readReq)
+			val identifiers = new util.ArrayList[String](){{
```

Comment:
this will lead to a new list for each iteration. So every DataNode.list call will happen with single identifier, which is wrong

Reply:
Related to the previous review comment about the need for Response object in the System Update call. Please clarify this confusion regarding that function prototype.